simplefs/src/fs.rs

```diff
@@ -0,0 +1,150 @@
+use std::collections::BTreeMap;
+use std::path::Path;
+
+use crate::alloc::Bitmap;
+use crate::io::BlockStorage;
+use crate::sb::SuperBlock;
+
+use thiserror::Error;
+use zerocopy::{AsBytes, FromBytes};
+
+pub const BLOCK_SIZE: usize = 4096;
+
+const SB_MAGIC: u32 = 0x5346_5342; // SFSB
+
+const NODE_SIZE: usize = 256;
+
+const FT_UNKNOWN: u8 = 0;
+const FT_FILE: u8 = 1;
+const FT_DIR: u8 = 2;
+
+#[repr(packed)]
+#[derive(AsBytes, FromBytes)]
+pub struct Inode {
+    /// The absolute postition of the inode in the filesystem.
+    inumber: u32,
+    /// The file mode (e.g full access - drwxrwxrwx).
+    mode: u16,
+    /// The total size in bytes.
+    size: u32,
```

Comment:
i've been overly achored to support for Large files. i think this is good starting out and i'm just gonna hang a reminder on it to come back and figure out for when we wanna store a folder with like every season of the Office